classes/services/PKPNavigationMenuService.inc.php

```diff
@@ -359,9 +354,91 @@ public function getMenuTree(&$navigationMenu) {
 			}
 		}
 
-		foreach($items as $item) {
-			$this->getDisplayStatus($item, $navigationMenu);
+		$navigationMenuDao = \DAORegistry::getDAO('NavigationMenuDAO');
+		$cache = $navigationMenuDao->getCache($navigationMenu->getId());
+		$json = json_encode($navigationMenu);
+		$cache->setEntireCache($json);
+	}
+
+
+
+	/**
+	 * Get a tree of NavigationMenuItems assigned to this menu
+	 * @param $navigationMenu \NavigationMenu
+	 *
+	 * @return array Hierarchical array of menu items
+	 */
+	public function getMenuTree(&$navigationMenu) {
```

Comment:
I don't think the reference is required.

Reply:
I think re reference is needed. The navigationMenu needs to "be changed" so that its menuTree array property is populated. (after the call the same navigationMenu is being used by the code)

Reply:
OK, thanks!